api/apps/api/src/modules/scenarios/run.service.spec.ts

```diff
@@ -0,0 +1,252 @@
+        scenarioId: 'scenario-1',
+      },
+      isActive: async () => true,
+      isWaiting: async () => false,
+      updateProgress: jest.fn(),
+    },
+    waitingJob: {
+      data: {
+        scenarioId: 'scenario-1',
+      },
+      isActive: async () => false,
+      isWaiting: async () => true,
+      remove: jest.fn(),
+    },
+    notCancelableJob: {
+      data: {
+        scenarioId: 'scenario-1',
+      },
+      isActive: async () => false,
+      isWaiting: async () => false,
+    },
+    otherJob: {
+      data: {
+        scenarioId: 'other-scenario',
+      },
+    },
+    getRunService() {
+      return testingModule.get(RunService);
+    },
+    schedulingJob() {
+      this.creatingEvents();
+      fakeQueue.add.mockImplementation(() => {
+        //
+      });
+      fakeScenarioRepo.update.mockImplementation(() => {
+        //
+      });
+    },
+    creatingEvents() {
+      fakeApiEvents.create.mockImplementation(() => {
+        //
+      });
+    },
+    GivenActiveJob() {
+      const jobs = [this.otherJob, this.activeJob];
+      this.fakeQueue.getJobs.mockImplementation((...args) => {
+        expect(args).toStrictEqual([['active', 'waiting']]);
+        return jobs;
+      });
+    },
+    GivenWaitingJob() {
+      const jobs = [this.otherJob, this.waitingJob];
+      this.fakeQueue.getJobs.mockImplementation((...args) => {
+        expect(args).toStrictEqual([['active', 'waiting']]);
+        return jobs;
+      });
+    },
+    GivenNoCancellableJob() {
+      const jobs = [this.otherJob, this.notCancelableJob];
+      this.fakeQueue.getJobs.mockImplementation((...args) => {
+        expect(args).toStrictEqual([['active', 'waiting']]);
+        return jobs;
+      });
+    },
+    ThenProgressOfActiveJobIsSetToCancel() {
+      expect(fixtures.activeJob.updateProgress).toBeCalledTimes(1);
+      expect(fixtures.activeJob.updateProgress).toBeCalledWith({
+        canceled: true,
+      });
+    },
+    ThenShouldUpdateScenario() {
+      expect(fixtures.fakeScenarioRepo.update).toBeCalledTimes(1);
+      expect(fixtures.fakeScenarioRepo.update).toBeCalledWith(`scenario-1`, {
+        ranAtLeastOnce: true,
+      });
+    },
+    ThenShouldEmitSubmittedEvent() {
+      expect(fixtures.fakeApiEvents.create).toBeCalledTimes(1);
+      expect(fixtures.fakeApiEvents.create).toBeCalledWith({
+        topic: `scenario-1`,
+        kind: API_EVENT_KINDS.scenario__run__submitted__v1__alpha1,
+      });
+    },
+    ThenShouldAddJob() {
+      expect(fixtures.fakeQueue.add).toBeCalledTimes(1);
+      expect(fixtures.fakeQueue.add).toBeCalledWith(`run-scenario`, {
+        scenarioId: `scenario-1`,
+      });
+    },
+    async ThenEventCreated(kind: API_EVENT_KINDS) {
+      await waitForExpect(() => {
+        expect(fixtures.fakeApiEvents.create).toBeCalledTimes(1);
+        expect(fixtures.fakeApiEvents.create).toBeCalledWith({
+          kind,
+          topic: `scenario-1`,
+        });
+      }, 100);
```

Comment:
because it's made through an event emitter, so the code jumps out to the event loop, and event processing is done asynchronously without any promise to await

Reply:
got rid of the passed interval as the default is sufficient in 7e05c206